src/test/ui/const-unsized.stderr

```diff
@@ -16,13 +16,13 @@ LL | const CONST_FOO: str = *"foo";
    = help: the trait `std::marker::Sized` is not implemented for `str`
    = note: constant expressions must have a statically known size
 
-error[E0277]: the trait bound `std::fmt::Debug + std::marker::Sync + 'static: std::marker::Sized` is not satisfied
+error[E0277]: the trait bound `dyn std::fmt::Debug + std::marker::Sync + 'static: std::marker::Sized` is not satisfied
```

Comment:
I have to admit — tracing through ppaux.rs — I couldn't quite tel how this arises. Can somebody clarify for me? I imagine that this message is produced from `traits/error_reporting.rs`:

https://github.com/rust-lang/rust/blob/90f34b5f83d8fb0dd39de7a6d7cfac4d10cb2218/src/librustc/traits/error_reporting.rs#L598-L601

But it seems like that is printing a `PolyTraitRef`, and I'm not clear on the path that leads from that code to the `TyDynamic` code. What am I missing?

Reply:
The `TyDynamic` is the `Self` type, printed by this:
https://github.com/rust-lang/rust/blob/4a9c58c6bd0e609538053a35685e206f66019820/src/librustc/util/ppaux.rs#L1258

Reply:
Oh; in that case, this seems...correct.

Reply:
it might be nicer to print "the trait `Sized` is not implemented for `dyn ...`", like we used to, but that seems separate